### PR TITLE
Decompile KaleidoScope_Draw

### PR DESCRIPTION
**Pull request by rogual**

### CODE REVIEW

File: src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c [hunk 1]
```diff
@@ -76,7 +79,54 @@
...
+    }
+    if (pCtx->state > 10 && pCtx->state <= 17) {
+        func_80825C14(globalCtx);
+    }
+    if (pCtx->flag == 1 || pCtx->flag == 2) {
+        func_808161AC(globalCtx);
+    }
+
+    /* "../z_kaleido_map_PAL.c" */
+    Graph_CloseDisps(dispRefs, globalCtx->state.gfxCtx, D_8082FD94, 3254);
```
Comment (fig02) [inline src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c:128]: you can put the string directly in there, you dont need to reference it by symbol

File: src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c [hunk 2]
```diff
@@ -76,7 +79,54 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_kaleido_scope/func_80825C14.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_kaleido_scope/KaleidoScope_Draw.s")
+extern const char D_8082FD78[], D_8082FD94[];
+
+void KaleidoScope_Draw(GlobalContext* globalCtx) {
+    Input* input = &globalCtx->state.input[0];
+    PauseContext* pCtx = &globalCtx->pauseCtx;
+    InterfaceContext* iCtx = &globalCtx->interfaceCtx;
```
Comment (fig02) [inline src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c:87]: if these temps are necessary, I would go with the full name of the context to make it easier to read

Comment (fig02) [reply src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c:87]: i didnt mean to highlight input there.
pCtx -> pauseCtx
iCtx -> interfaceCtx

File: src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c [hunk 3]
```diff
@@ -76,7 +79,54 @@
...
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
+
+    /* "../z_kaleido_map_PAL.c" */
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, D_8082FD78, 3188);
```
Comment (fig02) [inline src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c:92]: here too, strings can just can be added directly

File: src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c [hunk 4]
```diff
@@ -76,7 +79,54 @@
...
+
+    pCtx->inputX = (s16)input->rel.in.x;
+    pCtx->inputY = (s16)input->rel.in.y;
```
Comment (fig02) [inline src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c:95]: are these casts neccessary?

Comment (rogual) [reply src/overlays/actors/ovl_kaleido_scope/z_kaleido_scope.c]: Seems not, nice

**Pull request merged**